ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -1242,22 +1242,17 @@ extension SignalProducerType {
 
 			if shouldStartUnderlyingProducer {
 				self.producer
-					.takeUntil(token.willDeallocSignal)
+					.takeUntil(tracker.deallocSignal)
 					.start(initializedObserver)
 			}
 		}
 	}
 }
 
-private extension NSObject {
-	var willDeallocSignal: SignalProducer<(), NoError> {
-		return self
-			.rac_willDeallocSignal()
-			.toSignalProducer()
-			.map { _ in () }
-			.mapError { error in
-				fatalError("Unexpected error: \(error)")
-				()
-			}
+private final class Tracker {
+	let (deallocSignal, observer) = Signal<(), NoError>.pipe()
+
```

Comment:
`private` modifier in Swift 2.2 means the symbol is visible within the current file, not within the current scope. So adding `private` to `observer` property have no effect. This should be changed in Swift 3 by the proposal: https://github.com/apple/swift-evolution/blob/master/proposals/0025-scoped-access-level.md